packages/utils/src/lib/execute-process.ts

```diff
@@ -133,33 +139,38 @@ export type ProcessObserver = {
  * @param cfg - see {@link ProcessConfig}
  */
 export function executeProcess(cfg: ProcessConfig): Promise<ProcessResult> {
-  const { observer, cwd, command, args, ignoreExitCode = false } = cfg;
-  const { onStdout, onError, onComplete } = observer ?? {};
+  const { command, args, observer, ignoreExitCode = false, ...options } = cfg;
+  const { onStdout, onStderr, onError, onComplete } = observer ?? {};
   const date = new Date().toISOString();
   const start = performance.now();
 
   return new Promise((resolve, reject) => {
     // shell:true tells Windows to use shell command for spawning a child process
-    const process = spawn(command, args, { cwd, shell: true });
+    const spawnedProcess = spawn(command, args ?? [], {
+      shell: true,
+      ...options,
+    }) as ChildProcessByStdio<Writable, Readable, Readable>;
```

Comment:
:warning: This change introduces a potential safety hole, the `stdout` and `stderr` streams can be made `null` if `stdio` is set to anything other than `'pipe'` (previously this wasn't exposed). 

![image](https://github.com/user-attachments/assets/5fdf130a-557a-4513-b374-c38d8827ba1e)

If you want to enable callers to customizes the streams, then you should make sure it won't result in a runtime error:

![image](https://github.com/user-attachments/assets/a654d95f-fa26-41e8-a0ac-55eca8034a99)

If you want to restrict the caller to only the default `stdout`/`stderr` streams, then you should tighten the type definitions:

![image](https://github.com/user-attachments/assets/9f454149-d1ae-497b-a73d-1034628b7a4f)

I would go with the 2nd option, since `executeProcess` is unlikely to be useful for custom `stdio` streams.

Reply:
went with second option and added the restricted type to the options.